sky/data/storage.py

```diff
@@ -1865,10 +1902,615 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+            # If is_sky_managed is specified, then we take no action.
+            self.is_sky_managed = is_new_bucket
+
+    def _init_storage_account_and_resource_group(self) -> None:
+        """Initialize storage account and resource group.
+
+        Checks if the set resource group name and storage account name already
+        exists under user's subscription ID. If not, creates them and assigns
+        Storage Blob Data Owner role of the storage account.
+
+        Raises:
+            StorageBucketCreateError: If storage account attempted to be
+                created already exists
+        """
+        try:
+            # obtains detailed information about resource group under the
+            # user's subscription. Used to check if the name already exists
+            self.resource_client.resource_groups.get(
+                self.resource_group_name)
+        except azure.exceptions().ResourceNotFoundError as e:
+            if 'Code: ResourceGroupNotFound' in e.message:
+                with rich_utils.safe_status(
+                    f'[bold cyan]Setting up:\n'
+                    f'- Resource group: {self.resource_group_name}'):
+                    self.resource_client.resource_groups.create_or_update(
+                        self.resource_group_name, {'location': self.region})
+        # check if the storage account name already exists under the
+        # given resource group name.
+        try:
+            self.storage_client.storage_accounts.get_properties(
+                self.resource_group_name, self.storage_account_name)
+        except azure.exceptions().ResourceNotFoundError as e:
+            if 'Code: ResourceNotFound' in e.message:
+                with rich_utils.safe_status(
+                    f'[bold cyan]Setting up:\n'
+                    f'- Storage account: {self.storage_account_name}'):
+                    try:
+                        creation_response = (
+                            self.storage_client.storage_accounts.begin_create(
+                                self.resource_group_name,
+                                self.storage_account_name, {
+                                    'sku': {
+                                        'name': 'Standard_GRS'
+                                    },
+                                    'kind': 'StorageV2',
+                                    'location': self.region,
+                                    'encryption': {
+                                        'services': {
+                                            'blob': {
+                                                'key_type': 'Account',
+                                                'enabled': True
+                                            }
+                                        },
+                                        'key_source': 'Microsoft.Storage'
+                                    },
+                                }).result())
```

Comment:
I'm not sure if here is the best place to put those code.. How about refactor it into `sky/provision/azure/instance.py`? feeling like this should be put into the provisioner module. Not sure, could discuss w/ the group.

Reply:
This `begin_create` method isn't used anywhere else and it's pretty self-explanatory on what it does. I believe moving this to `sky/provision/azure/instance.py` would cause more complexity unless it becomes general enough that it's called from other places as well.

Reply:
My opinion is that this seems like some abstraction leak, i.e. ideally we should put all Azure API function calls in a single module, both storage and instance related ones. But open to discussion as well :)) Lets see other ppl's opinions.